### PR TITLE
Fix JENKINS-34216

### PR DESCRIPTION
**Pull request by joshuaspence**
Fixes JENKINS-34216 by following recommendations from aws/aws-sdk-java#427.

### CODE REVIEW

**Comment by joshuaspence** [comment]
To test this, I created the following job:

```
<project>
    <description>A test for T29479.</description>
    <keepDependencies>false</keepDependencies>
    <properties></properties>
    <scm class='hudson.scm.NullSCM'></scm>
    <canRoam>true</canRoam>
    <disabled>false</disabled>
    <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
    <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
    <triggers class='vector'></triggers>
    <concurrentBuild>true</concurrentBuild>
    <builders>
        <hudson.tasks.Shell>
            <command>dd if=/dev/zero of=large-file bs=1M count=1024</command>
        </hudson.tasks.Shell>
    </builders>
    <publishers>
        <hudson.plugins.chucknorris.CordellWalkerRecorder>
            <factGenerator></factGenerator>
        </hudson.plugins.chucknorris.CordellWalkerRecorder>
        <hudson.plugins.s3.S3BucketPublisher>
            <profileName>REDACTED</profileName>
            <entries>
                <hudson.plugins.s3.Entry>
                    <sourceFile>large-file</sourceFile>
                    <bucket>REDACTED</bucket>
                    <storageClass>STANDARD</storageClass>
                    <selectedRegion>us-east-1</selectedRegion>
                    <noUploadOnFailure>true</noUploadOnFailure>
                    <uploadFromSlave>false</uploadFromSlave>
                    <managedArtifacts>false</managedArtifacts>
                    <useServerSideEncryption>false</useServerSideEncryption>
                    <flatten>false</flatten>
                </hudson.plugins.s3.Entry>
            </entries>
            <userMetadata></userMetadata>
        </hudson.plugins.s3.S3BucketPublisher>
    </publishers>
    <buildWrappers>
        <EnvInjectPasswordWrapper>
            <injectGlobalPasswords>true</injectGlobalPasswords>
            <maskPasswordParameters>true</maskPasswordParameters>
            <passwordEntries></passwordEntries>
        </EnvInjectPasswordWrapper>
        <org.jenkinsci.plugins.builduser.BuildUser></org.jenkinsci.plugins.builduser.BuildUser>
        <hudson.plugins.ansicolor.AnsiColorBuildWrapper>
            <colorMapName>xterm</colorMapName>
        </hudson.plugins.ansicolor.AnsiColorBuildWrapper>
        <com.michelin.cio.hudson.plugins.maskpasswords.MaskPasswordsBuildWrapper></com.michelin.cio.hudson.plugins.maskpasswords.MaskPasswordsBuildWrapper>
        <hudson.plugins.timestamper.TimestamperBuildWrapper></hudson.plugins.timestamper.TimestamperBuildWrapper>
    </buildWrappers>
    <logRotator>
        <daysToKeep>-1</daysToKeep>
        <numToKeep>-1</numToKeep>
        <artifactDaysToKeep>30</artifactDaysToKeep>
        <artifactNumToKeep>100</artifactNumToKeep>
    </logRotator>
    <scmCheckoutRetryCount>3</scmCheckoutRetryCount>
    <quietPeriod>5</quietPeriod>
</project>
```

Prior to my patch, this job failed 40% of the time (4 out of 10 builds).

File: src/main/java/hudson/plugins/s3/Uploads.java [hunk 1]
```diff
@@ -20,7 +21,14 @@ private Uploads() {}
     private final transient HashMap<FilePath, InputStream> openedStreams = new HashMap<>();
 
     public void startUploading(TransferManager manager, FilePath file, InputStream inputsStream, String bucketName, String objectName, ObjectMetadata metadata) throws AmazonServiceException {
-        final Upload upload = manager.upload(bucketName, objectName, inputsStream, metadata);
+        final PutObjectRequest request = new PutObjectRequest(bucketName, objectName, inputsStream, metadata);
+
+        // Set the buffer size (ReadLimit) equal to the multipart upload size,
+        // allowing us to resend data if the connection breaks.
+        request.getRequestClientOptions().setReadLimit(10485760);
```
Comment (Jimilian) [inline src/main/java/hudson/plugins/s3/Uploads.java]: Please, extract a constant variable for this value. And it's not clear why 10485760?

**Comment by Jimilian** [comment]
@joshuaspence, thanks for your PR! Sorry, that didn't mention it earlier - I didn't receive update by some reason. Please, fix tiny code style issue. After that I will merge your pull request and will release new version.

File: src/main/java/hudson/plugins/s3/Uploads.java (hunk 1 above)
Comment (joshuaspence) [inline src/main/java/hudson/plugins/s3/Uploads.java]: Yeah, will do. As for the specific value, I'll be honest in saying that I am not entirely sure... I just followed the advice given at aws/aws-sdk-java#427.

**Comment by joshuaspence** [comment]
I believe that I have addressed your feedback.

**Comment by Jimilian** [comment]
@joshuaspence Compilation?

**Comment by joshuaspence** [comment]
Should be good now.

**Comment by Jimilian** [comment]
Great!

**Pull request merged**